src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1651,6 +1651,11 @@ void adjustIdentifierMapIfAny(SimpleNameSyntax node, bool invoked)
                     Binder current = this;
                     while (current is not (null or InMethodBinder { IdentifierMap: not null }))
                     {
+                        if (current is ContextualAttributeBinder)
```

Comment:
>if (current is ContextualAttributeBinder) [](http://example.com/codeflow?start=24&length=41)

If we end up keeping an additional logic here (which wouldn't be necessary with the alternative approach), I would prefer it to be a syntactic check based on `id`.